.github/workflows/build_and_run_tests.yml

```diff
@@ -0,0 +1,68 @@
+# Test changes that have been pushed to the master branch
+
+name: Testing and Validation
+
+# Controls when the action will run.
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+#env:
+  # Obtained from https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
+  # TODO: add something later that can check the runners for hyperthreading
+  #LINUX_NUM_PROC_CORES: 2
+  #MACOS_NUM_PROC_CORES: 3
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Run general unit tests in linux environment
+  test_unit:
+    # The type of runner that the job will run on
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    - name: Checkout the commit
+      uses: actions/checkout@v4
+    
+   # - name: check gcc version
+   #   run: gcc --version
+    
+    - name: Build and Run Standalone
+      run: |
+        mkdir build && cd build
+        cmake ../ -DSTANDALONE=ON
+        make && cd ..
+        ./build/lasam_standalone configs/config_lasam_Phillipsburg.txt
+        ./build/lasam_standalone configs/config_lasam_Bushland.txt
```

Comment:
can you please move this to the end, 1. unittest, 2. synthetic tests, 3. real tests (just to be consistent)

Reply:
Agree on the first point. Several submodules have moved the build directory to the next sub-directory. So that's probably what need to be done.

Working on re-ordering the tests in the build_and_run_tests.